applications/GeoMechanicsApplication/custom_retention/retention_law.h

```diff
@@ -91,23 +79,41 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) RetentionLaw
 
         ~Parameters() = default;
 
-        void SetFluidPressure   (const double rFluidPressure)    { mFluidPressure = rFluidPressure; };
+        void SetFluidPressure (double rFluidPressure)
+        {
+            mFluidPressure = rFluidPressure;
+            mFluidPressureIsSet = true;
+        };
 
-        double GetFluidPressure()    const { return mFluidPressure;    }
+        double GetFluidPressure() const
+        {
+            KRATOS_ERROR_IF_NOT(mFluidPressureIsSet) << "Fluid pressure is not yet set in the retention "
+                                                        "law when trying to retrieve it, aborting.\n";
+            return mFluidPressure;
+        }
 
         const ProcessInfo &GetProcessInfo() const
         {
             return mrCurrentProcessInfo;
         }
+
         const Properties &GetMaterialProperties() const
         {
             return mrMaterialProperties;
         }
+
         const GeometryType &GetElementGeometry() const
         {
             return mrElementGeometry;
         }
 
+    private:
+        double mFluidPressure = 0.0;
+        bool mFluidPressureIsSet = false;
```

Comment:
I would model a variable that may have an "unset" state using `std::optional`:
```suggestion
        std::optional<double> mFluidPressure;
```

Reply:
That is better.